packages/blocks/src/store/private-actions.js

```diff
@@ -0,0 +1,39 @@
+/**
+ * Internal dependencies
+ */
+import { processBlockType } from './process-block-type';
+
+/** @typedef {import('../api/registration').WPBlockType} WPBlockType */
+
+/**
+ * Add block type metadata to the store.
+ *
+ * @param {string}      name      Block name.
+ * @param {WPBlockType} blockType Block type metadata.
+ */
+export function addBootstrappedBlock( name, blockType ) {
+	return {
+		type: 'ADD_BOOTSTRAPPED_BLOCK_TYPE',
+		name,
+		blockType,
+	};
+}
+
+/**
+ * Add unprocessed block type settings to the store.
+ *
+ * @param {string}      name      Block name.
+ * @param {WPBlockType} blockType Unprocessed block type settings.
+ */
+export const addUnprocessedBlock =
+	( name, blockType ) =>
+	( { dispatch } ) => {
+		dispatch( { type: 'ADD_UNPROCESSED_BLOCK_TYPE', name, blockType } );
+		const processedBlockType = dispatch(
```

Comment:
As far as I can tell `processBlockType` doesn't return any action to dispatch. Does it really need to be wrapped with the `dispatch` call?

Reply:
`processBlockType` is a thunk action. It can be dispatched and will return the `processedBlockType` result. The good reason to write it like this is that the thunk gets access to `select` and `dispatch` automatically.

Reply:
I figured out it could be dispatched now. I was surprised to learn it is a way to go about it. Interestingly, I shaped it in a manner compatible with `dispatch` from day one. Thank you for sharing the rationale; it is very insightful.